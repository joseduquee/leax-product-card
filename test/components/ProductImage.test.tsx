import renderer from 'react-test-renderer';
import { ProductImage } from '../../src/components/ProductImage';
import { product2 } from '../data/products';
import React from 'react';


describe('ProductImage', () => {
    
    test('Debe crear el componente correctamente con la imagen personalizada', () => {
        
        const wrapper = renderer.create(
            <ProductImage img="https://definicion.de/wp-content/uploads/2009/12/paisaje-1.jpg" /> 
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    });

    test('Debe crear el componente correctamente con la imagen del producto', () => {
        
        const wrapper = renderer.create(
            <ProductImage img={ product2.img } /> 
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    });
    

})
