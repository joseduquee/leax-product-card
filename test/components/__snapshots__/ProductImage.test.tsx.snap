// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe crear el componente correctamente con la imagen 1`] = `
<img
  alt="product img"
  className="productCard undefined"
  src="./coffee-mug.png"
/>
`;

exports[`ProductImage Debe crear el componente correctamente con la imagen del producto 1`] = `
<img
  alt="product img"
  className="productCard undefined"
  src="./coffee-mug.png"
/>
`;

exports[`ProductImage Debe crear el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="product img"
  className="productCard undefined"
  src="https://definicion.de/wp-content/uploads/2009/12/paisaje-1.jpg"
/>
`;
